src/app/modules/item/services/item-task-answer.service.ts

```diff
@@ -102,18 +105,46 @@ export class ItemTaskAnswerService implements OnDestroy {
     );
 
     // Step 4: Save grade in backend
-    return combineLatest([ this.taskToken$, answerToken$, grade$ ]).pipe(
+    const saveGrade$ = combineLatest([ this.taskToken$, answerToken$, grade$ ]).pipe(
       take(1),
       switchMap(([ taskToken, answerToken, grade ]) => this.gradeService.save(
         taskToken,
         answerToken,
         grade.score,
         grade.scoreToken ?? undefined,
       )),
+      shareReplay(1),
     );
+    combineLatest([ grade$, saveGrade$ ]).subscribe(([ grade ]) => {
+      if (grade.score !== undefined) this.updateCurrentContentScore(grade.score);
+    });
+    return saveGrade$;
   }
 
   clearAnswer(): Observable<unknown> {
     return this.task$.pipe(take(1), switchMap(task => task.reloadAnswer('')));
   }
+
+  private updateCurrentContentScore(currentScore: number): void {
+    const content = this.currentContent.current();
+    if (!content) throw new Error('unexpected: no content to update');
+    if (!isItemInfo(content)) throw new Error('unexpected: content is not item info');
+
```

Comment:
Mmh no.
For the same reason which was described above.
Ok this is conceptual. But in practice, that two different parts push on the same channel at the same time... item-by-id needs to know about the score! If item-by-id repush something for some reason (we want to re-add navigation in item-by-id for instance ;-)), it shouldn't break this score update for instance.